server/balancer.go

```diff
@@ -370,3 +372,207 @@ func (r *replicaChecker) checkBestReplacement(region *RegionInfo) Operator {
 	}
 	return newTransferPeer(region, oldPeer, newPeer)
 }
+
+// RegionStateValue record each hot region state
+type RegionStateValue struct {
+	RegionID       uint64    `json:"region_id"`
+	WrittenBytes   uint64    `json:"written_bytes"`
+	UpdateCount    int       `json:"update_count"`
+	LastUpdateTime time.Time `json:"last_update_time"`
+	StoreID        uint64    `json:"-"`
+	antiCount      int
+	version        uint64
+}
+
+// MetaRegionStatus is a list of a group region state type
+type MetaRegionStatus []RegionStateValue
+
+func (m MetaRegionStatus) Len() int           { return len(m) }
+func (m MetaRegionStatus) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+func (m MetaRegionStatus) Less(i, j int) bool { return m[i].WrittenBytes > m[j].WrittenBytes }
+
+// RegionsHotStatus record all hot regions in one store with sequence
+type RegionsHotStatus struct {
+	TotalWrittenBytes uint64           `json:"total_written"`
+	MetaStatus        MetaRegionStatus `json:"status"`
+}
+
+type balanceHotRegionScheduler struct {
+	opt         *scheduleOption
+	limit       uint64
+	scoreStatus map[uint64]RegionsHotStatus // store id -> regions status in this store
+	r           *rand.Rand
+}
+
+func newBalanceHotRegionScheduler(opt *scheduleOption) *balanceHotRegionScheduler {
+	return &balanceHotRegionScheduler{
+		opt:         opt,
+		limit:       1,
+		scoreStatus: make(map[uint64]RegionsHotStatus),
+		r:           rand.New(rand.NewSource(time.Now().UnixNano())),
+	}
+}
+
+func (l *balanceHotRegionScheduler) GetName() string {
+	return "balance-hot-region-scheduler"
+}
+
+func (l *balanceHotRegionScheduler) GetResourceKind() ResourceKind {
+	return priorityKind
+}
+
+func (l *balanceHotRegionScheduler) GetResourceLimit() uint64 {
+	return l.limit
+}
+
+func (l *balanceHotRegionScheduler) Prepare(cluster *clusterInfo) error { return nil }
+
+func (l *balanceHotRegionScheduler) Cleanup(cluster *clusterInfo) {}
+
+func (l *balanceHotRegionScheduler) Schedule(cluster *clusterInfo) Operator {
+	log.Info("")
+	l.CalculateScore(cluster)
+	region, leader := l.SelectTransferLeader(cluster)
+	if region == nil {
```

Comment:
It is means if the transferLeader policy not work, transferPeer will not work too, why?